### ISSUE TITLE
No Ruby 3 support

### ISSUE THREAD

**Issue by darkBuddha** [paypal/paypalhttp_ruby#12]
### General information

* SDK/Library version: 1.0.1
* Ruby Version: 3.0.2

### Issue description

The gem should support the latest Ruby version. It offers major advantages over Ruby 2.

**Comment by Sexual** [comment]
One potential simple fix?: [https://github.com/rollbar/rollbar-gem/pull/988/files](https://github.com/rollbar/rollbar-gem/pull/988/files)

**Comment by dorianmariefr** [comment]
See #18 

or a quick fix:

```ruby
module URI
  # for PayPal client
  def self.escape(string)
    URI.encode_www_form_component(string).gsub('+', '%20')
  end
end
```

**Comment by darkBuddha** [comment]
Jesus. Not cool, PayPal. With Ruby 3.1 and Rails 7 there is some added urgency.

**Comment by plerohellec** [comment]
Any chance of getting this fixed now that ruby 3.1 is out?

**Comment by darkBuddha** [comment]
> Any chance of getting this fixed now that ruby 3.1 is out?

https://stripe.com/docs/api?lang=ruby

**Comment by dorianmariefr** [comment]
stripe doesn't have paypal support ;)

**Comment by darkBuddha** [comment]
> stripe doesn't have paypal support ;)

Stripe doesn't block developers (customers) from upgrading their projects...

**Comment by dorianmariefr** [comment]
you can use the monkey patch or my branch in the fork, or make your own fork. I agree PayPal is technically horrible (the developer docs are the worst website on the internet) but hey people like to use paypal (I do)

**Comment by darkBuddha** [comment]
> you can use the monkey patch or my branch in the fork, or make your own fork. I agree PayPal is technically horrible (the developer docs are the worst website on the internet) but hey people like to use paypal (I do)

Thanks man, great work!

**Comment by crookedneighbor** [comment]
This is fixed in v2.0.0, which was just released.

**Comment by xeruf** [comment]
since we can't rely on trust on such a sensitive area as payments, pin the commit on the fork ;)

    gem 'paypal-checkout-sdk', git: 'https://github.com/darkBuddha/Checkout-Ruby-SDK.git', :ref => 'f1aab0f4e367aad1de937bba6cbc4abb22900339'
